simpleviewinc/keg-components
# ZEN-317: Vector Icon Removal

**Ticket**: [ZEN-317](https://jira.simpleviewtools.com/browse/ZEN-317)

## Context

* when building the tap for export, vector-icons is always in the bundle, even when all components that use it in keg-components are removed. It is not being tree-shaken out.
* furthermore, it's arguable that `@expo/vector-icons` doesn't really belong in `keg-components`, but in the apps 

## Goal

* remove the use of `@expo/vector-icons` from keg-components
* update the components to work without them

## Updates

* `configs/rollup.config.js`
  * removed vector-icons from peer dependencies, since it is no longer actually required
* `package.json`
  * incremented the version
* `src/components/appHeader/appHeader.js`
  * updated to let the consumer pass-in their own icon set, to work with the icon name props
* `src/components/icon/icon.*.js`
  * removed imports from `@expo/vector-icons`, which were only being used for default values
* `src/components/**/*.stories.js`
  * updated stories to import `@expo/vector-icons` and pass them in to the icon to keep our stories the same and demonstrate the expected consumer calling context
* `src/components/indicator/indicator.web.js`
  * updated to allow user to pass in the icon element
  * this isn't really necessary, b/c we aren't using `.web.js`, but I did it for consistency
* `src/**/__tests__/*.js`
  * updated tests to work with changes

## Testing

* run the docker container: `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:vector-icons-removal`
* verify that the  `AppHeader`, `Icon`, and `Loading` components all look and behave like they did before
* attach to the container (run `docker exec -it <container_id> /bin/bash`)
* `yarn build`
* `head -10 ./build/esm/kegComponents.native.js`
* verify that you see no imports for `@expo/vector-icons`
